src/cpp/transport/UDPTransportInterface.cpp

```diff
@@ -393,6 +396,25 @@ bool UDPTransportInterface::send(
 
         try
         {
+#ifndef _WIN32
+            if(timeout.count() > 0)
+            {
+                struct timeval timeStruct;
+                timeStruct.tv_sec = 0;
+                timeStruct.tv_usec = timeout.count();
+                setsockopt(getSocketPtr(socket)->native_handle(), SOL_SOCKET, SO_SNDTIMEO,
+                        reinterpret_cast<const char*>(&timeStruct), sizeof(timeStruct));
+            }
+            else
+            {
+                struct timeval timeStruct;
+                timeStruct.tv_sec = 0;
+                timeStruct.tv_usec = 0;
```

Comment:
I think both branches can be merged with the following change:
```suggestion
                timeStruct.tv_usec = timeout.count() > 0 ? timeout.count() : 0;
```

Reply:
Done